src/artifacts_poetry_plugin/plugin.py

```diff
@@ -9,143 +9,234 @@
-                project_package.files.append(file_data)
-    packages.append(project_package)
-
-
-def get_deploy_packages(poetry, cache_dir):
-    """Gets all deployable packages from the given cache directory. A package is deployable if it has any wheel files associated with it"""
-    packages = get_packages(poetry)
-    wheel_files = get_cached_wheel_files(cache_dir)
-    deployable_packages, non_deployable_packages = add_package_paths(
-        packages=packages, wheel_files=wheel_files
-    )
-    add_project_package(poetry, deployable_packages)
-    return deployable_packages, non_deployable_packages
-
-
-class ArtifactsDeploy(Command):
+            raise RuntimeError(err_msg)
+        if username is None or password is None:
+            raise RuntimeError(err_msg)
+        return RepoData(url, username, password)
+
+    def get_poetry_packages(self, poetry) -> List[Package]:
+        """Gets packages from the given poetry.lock file"""
+        locker = poetry.locker
+        repository = locker.locked_repository()
+        return repository.packages
+
+    def get_cached_wheel_files(self, cache_dir):
+        """Gets all wheel file locations for the given cache directory"""
+        wheel_files = {}
+        for root, dirs, files in os.walk(cache_dir):
+            for file in files:
+                if file.endswith(".whl") or file.endswith(".tar.gz"):
+                    wheel_files[file] = Path(os.path.join(root, file))
+        return wheel_files
+
+    def get_deploy_non_deploy_packages(self, packages: List[Package], wheel_files):
+        """Adds the wheel file locations for each package's files"""
+        non_deployable_packages = []
+        deployable_packages = []
+        for package in packages:
+            file_found = False
+            for file in package.files:
+                if file["file"] in wheel_files:
+                    file_found = True
+                    file["url"] = wheel_files[file["file"]]
+            if file_found:
+                deployable_packages.append(package)
+            else:
+                non_deployable_packages.append(package)
+        return deployable_packages, non_deployable_packages
+
+    def upload_packages(self, poetry_files, url, username, password, io):
+        deploy_set = set()
+        for poetry in poetry_files:
+            uploader = Uploader(poetry, io)
+            uploader.auth(username=username, password=password)
+            session = uploader.make_session()
+            for package in poetry.deployable_packages:
+                uploader._package = package
+                for file in package.files:
+                    if "url" in file and file["url"] not in deploy_set:
+                        deploy_set.add(file["url"])
+                        uploader._upload_file(
+                            session, url, file=file["url"], skip_existing=True
+                        )
+
+    def get_project_package(self, poetry):
```

Comment:
the naming of this feels off. really, this function is passed a poetry object and grabs the poetry object's existing package + the packaged files found in its `dist/` directory. maybe the method name could be `get_project_package_dist_package()` or something to that effect.

Reply:
renamed to get_project_dist_package